src/components/userGroups/K8sObjectPermissions/K8sPermissionModal.tsx

```diff
@@ -0,0 +1,153 @@
+        resource: k8sPermission?.resource,
+        action: k8sPermission?.action || { value: ActionTypes.VIEW, label: ActionTypes.VIEW },
+    }
+}
+
+export default function K8sPermissionModal({ selectedPermissionAction, k8sPermission, setK8sPermission, close }) {
+    const [k8PermissionList, setPermissionList] = useState([getEmptyPermissionObject(0, k8sPermission)])
+    const [namespaceMapping, setNamespaceMapping] = useState<Record<number, OptionType[]>>()
+    const [apiGroupMapping, setApiGroupMapping] = useState<Record<number, OptionType[]>>()
+    const [kindMapping, setKindMapping] = useState<Record<number, OptionType[]>>()
+    const [objectMapping, setObjectMapping] = useState<Record<number, OptionType[]>>()
+    const [isDataFilled, setIsDataFilled] = useState<boolean>(true)
+
+    useEffect(() => {
+        const disbale = k8PermissionList
+            ? k8PermissionList.find((item) => item.resource === null || item.resource?.length === 0)
+            : true
+        setIsDataFilled(!!disbale)
+    }, [k8PermissionList])
+
+    const handleK8sPermission = (action: string, key?: number, data?: any) => {
+        switch (action) {
+            case 'add':
+                k8PermissionList.splice(0, 0, getEmptyPermissionObject(k8PermissionList.length))
+                break
+            case 'delete':
+                k8PermissionList.splice(key, 1)
+                break
+            case 'clone':
+                k8PermissionList.splice(0, 0, { ...k8PermissionList[key], key: k8PermissionList.length })
+                break
+            case 'edit':
+                k8PermissionList[key].cluster = data
+                break
+            case 'onClusterChange':
+                k8PermissionList[key].cluster = data
+                k8PermissionList[key].namespace = null
+                k8PermissionList[key].group = null
+                k8PermissionList[key].kind = null
+                k8PermissionList[key].resource = null
+                break
+            case 'onNamespaceChange':
+                k8PermissionList[key].namespace = data
+                k8PermissionList[key].group = null
+                k8PermissionList[key].kind = null
+                k8PermissionList[key].resource = null
+                break
+            case 'onApiGroupChange':
+                k8PermissionList[key].group = data
+                k8PermissionList[key].kind = null
+                k8PermissionList[key].resource = null
+                break
+            case 'onKindChange':
+                k8PermissionList[key].kind = data
+                k8PermissionList[key].resource = null
+                break
+            case 'onObjectChange':
+                k8PermissionList[key].resource = data
+                break
+            case 'onRoleChange':
+                k8PermissionList[key].action = data
+                break
+            default:
+                break
+        }
+
+        setPermissionList([...k8PermissionList])
```

Comment:
Create a local var at the top for k8PermissionList & use it instead of mutating the state.

Reply:
done